tun/Cargo.toml

```diff
@@ -11,12 +11,20 @@ socket2 = "0.4"
 tokio = { version = "1.28", features = [] }
 byteorder = "1.4"
 
+futures = { version = "0.3.28", optional = true }
+
+[features]
+async = ["tokio/full", "dep:futures"]
```

Comment:
We should only depend on the subset of tokio that we need, which might just be the `net` feature